src/util-lua-tls.c

```diff
@@ -73,39 +98,34 @@ static int GetCertNotBefore(lua_State *luastate, const Flow *f, int direction)
     if (connp->cert0_not_before == 0)
         return LuaCallbackError(luastate, "error: no certificate NotBefore");
 
-    int r = LuaPushInteger(luastate, connp->cert0_not_before);
-
-    return r;
+    return LuaPushInteger(luastate, connp->cert0_not_before);
 }
 
-static int TlsGetCertNotBefore(lua_State *luastate)
+static int LuaTlsGetServerCertNotBefore(lua_State *luastate)
 {
-    int r;
-
-    if (!(LuaStateNeedProto(luastate, ALPROTO_TLS)))
-        return LuaCallbackError(luastate, "error: protocol not tls");
-
-    int direction = LuaStateGetDirection(luastate);
+    struct LuaTls *s = (struct LuaTls *)lua_touserdata(luastate, 1);
+    if (s == NULL || s->state == NULL) {
+        LUA_ERROR("failed to get flow");
+    }
```

Comment:
I think I prefer something like:
```
struct LuaTls *s = (struct LuaTls *)lua_checkudata(luastate, 1, tls_state_mt);
```

This function never returns if the data is invalid. This is only likely to happen in a programming error, and the result is an uncatchable error that will be logged (well, it can be caught by the lua programming when wrapped in pcall).  So its more like an exception. It means the user doesn't need to worry about always checking for nil if nil doesn't make sense.

You can then skip the NULL checks afterwards as well.  As `s` will always be valid, and we know that `s` only is ever created with a valid state.